onnxruntime/core/providers/coreml/builders/impl/pad_op_builder.cc

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/common/safeint.h"
+#include "core/providers/common.h"
+#include "core/framework/tensorprotoutils.h"
+#include "core/providers/coreml/builders/helper.h"
+#include "core/providers/shared/utils/utils.h"
+#include "core/optimizer/initializer.h"
+
+#ifdef __APPLE__
+#include "core/providers/coreml/builders/model_builder.h"
+#endif
+#include "core/providers/coreml/builders/op_builder_factory.h"
+
+#include "base_op_builder.h"
+
+namespace onnxruntime {
+namespace coreml {
+
+class PadOpBuilder : public BaseOpBuilder {
+  // Add operator related
+#ifdef __APPLE__
+ public:
+  void AddInitializersToSkip(ModelBuilder& model_builder, const Node& node) const override;
+
+ private:
+  Status AddToModelBuilderImpl(ModelBuilder& model_builder, const Node& node,
+                               const logging::Logger& logger) const override;
+#endif
+
+  // Operator support related
+ private:
+  bool IsOpSupportedImpl(const Node& node, const OpBuilderInputParams& input_params,
+                         const logging::Logger& logger) const override;
+
+  int GetMinSupportedOpSet(const Node& /* node */) const override {
+    // Note: before Pad-11, inputs `pads` and `constant_value` were attributes
+    return 11;
+  }
+};
+
+// Add operator related
+
+#ifdef __APPLE__
+void PadOpBuilder::AddInitializersToSkip(ModelBuilder& model_builder, const Node& node) const {
+  model_builder.AddInitializerToSkip(node.InputDefs()[1]->Name());  //  pads
+  model_builder.AddInitializerToSkip(node.InputDefs()[2]->Name());  //  constant_value
+  if (node.InputDefs().size() > 3) {
+    model_builder.AddInitializerToSkip(node.InputDefs()[3]->Name());  // axes
+  }
+}
+
+Status PadOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder,
+                                           const Node& node,
+                                           const logging::Logger& logger) const {
+  std::unique_ptr<COREML_SPEC::NeuralNetworkLayer> layer = CreateNNLayer(model_builder, node);
+
+  auto* coreml_pad = layer->mutable_padding();
+  auto* constant_padding_type = coreml_pad->mutable_constant();  // CoreML::Specification::PaddingLayerParams_PaddingConstant
+
+  const auto& input_defs = node.InputDefs();
+  std::vector<int64_t> input_shape;
+  GetShape(*input_defs[0], input_shape, logger);
+  const auto input_rank = input_shape.size();
+
+  const auto& pads_tensor = *model_builder.GetInitializerTensors().at(input_defs[1]->Name());            // pads
+  const auto& constant_value_tensor = *model_builder.GetInitializerTensors().at(input_defs[2]->Name());  // constant_value
+
+  Initializer constant_value_raw_data_init(constant_value_tensor);
+  float constant_value = constant_value_raw_data_init.DataAsSpan<float>()[0];
+  constant_padding_type->set_value(constant_value);
+
+  Initializer pads_initializer_raw_data(pads_tensor);
```

Comment:
what does "raw data" mean in these names? I would think something like "pads_initializer" would be fine.